client/my-sites/theme/main.jsx

```diff
@@ -130,12 +133,17 @@ class ThemeSheet extends Component {
 		section: '',
 	};
 
+	state = {};
+
 	scrollToTop = () => {
 		window.scroll( 0, 0 );
 	};
 
 	componentDidMount() {
 		this.scrollToTop();
+		if ( getQueryArgs()?.[ 'sync-active-theme' ] === 'true' ) {
+			this.props.themeStartActivationSync( this.props.siteId, this.props.themeId );
+		}
```

Comment:
Might it be cleaner if we handled this via a prop that we inject from the controller in `client/my-sites/theme/controller.jsx` based on `context.query?.['sync-active-theme']`.